components/runtime/src/extensions/discord.rs

```diff
@@ -374,6 +375,23 @@ pub async fn op_discord_edit_message(
         .into())
 }
 
+#[op]
+pub async fn op_discord_crosspost_message(
+    state: Rc<RefCell<OpState>>,
+    channel_id: Id<ChannelMarker>,
+    message_id: Id<MessageMarker>,
+) -> Result<(), AnyError> {
+    get_rt_ctx(&state)
+        .discord_config
+        .client
+        .crosspost_message(channel_id, message_id)
+        .exec()
+        .await
+        .map_err(|err| handle_discord_error(&state, err));
+
+    Ok(());
+}
+
```

Comment:
We need a check here to make sure the channel is on the guild, otherwise you could crosspost messages on other guilds, you can do this by adding the following:

`get_guild_channel(&state, &rt_ctx, channel_id).await?;`

Reply:
👍